frappe/sessions.py

```diff
@@ -319,6 +320,7 @@ def get_session_data(self):
 
 		data = self.get_session_data_from_cache()
 		if not data:
+			self._update_in_cache = True
```

Comment:
This flag is kinda required because what's pushed in cache is not what's fetched from DB, there are more ops that happen on it. (mostly just wrapping it another dict with other info or some reason)

Reply:
Maybe time to refactor sessions :skull: